src/registry.ts

```diff
@@ -153,32 +155,61 @@ export async function getEntry(
     );
   }
 
-  // Use empty string if the data is empty.
-  let data = "";
+  // Convert the revision from a string to bigint.
+  const revision = BigInt(response.data.revision);
+  const signature = Buffer.from(hexToUint8Array(response.data.signature));
+
+  // Use empty array if the data is empty.
+  let data = new Uint8Array([]);
+  // Try verifying the returned data as raw bytes.
   if (response.data.data) {
-    data = uint8ArrayToStringUtf8(hexToUint8Array(response.data.data));
+    data = hexToUint8Array(response.data.data);
   }
   const signedEntry = {
     entry: {
       dataKey,
       data,
-      // Convert the revision from a string to bigint.
-      revision: BigInt(response.data.revision),
+      revision,
     },
-    signature: Buffer.from(hexToUint8Array(response.data.signature)),
+    signature,
   };
   if (
-    signedEntry &&
-    !sign.detached.verify(
+    sign.detached.verify(
       hashRegistryEntry(signedEntry.entry, opts.hashedDataKeyHex),
       new Uint8Array(signedEntry.signature),
       hexToUint8Array(publicKey)
     )
   ) {
-    throw new Error("could not verify signature from retrieved, signed registry entry -- possible corrupted entry");
+    return signedEntry;
   }
 
-  return signedEntry;
+  // Try verifying the returned data as a string.
```

Comment:
I'm not sure why we need to do this -- it seems like even when set with strings, the signature should have been made on the underlying bytes.

Reply:
Why not let registry entries be inherently uint8[] from now on, with any interpretation happening at higher levels? I'm worried we'll find odd edge cases where data _can_ resolve to coherent strings, but shouldn't unless there's something in the signature process that signs uint8[] differently than strings, so signature verification is dependent on type.